pkg/xds/server/callbacks/dataplane_lifecycle.go

```diff
@@ -251,8 +251,12 @@ func (d *DataplaneLifecycle) proxyConnectedToAnotherCP(
 		return false, errors.Wrap(err, "could not get insight to determine if we can delete proxy object")
 	}
 
-	sub := insight.GetSpec().(generic.Insight).GetLastSubscription().(*mesh_proto.DiscoverySubscription)
-	if sub != nil && sub.ControlPlaneInstanceId != d.cpInstanceID {
+	subs := insight.GetSpec().(generic.Insight).AllSubscriptions()
+	if len(subs) == 0 {
+		return false, nil
+	}
+
+	if sub := subs[len(subs)-1].(*mesh_proto.DiscoverySubscription); sub.ControlPlaneInstanceId != d.cpInstanceID {
```

Comment:
@jakubdyszkiewicz it's not clear to me whether we want to change this logic. It's not quite the same mistake as in the finalizer. Checking the last subscription here seems to just be a way to assign responsibility for deleting the `Dataplane` resource.

This idea seems unrelated to the online/offline management

Reply:
I'm not sure. Technically it shouldn't matter. I think it's fine either way but we can keep it as is to minimize changes

Reply:
I'm going to leave it. It's not clear to me how or why it should even work differently. What should the check even be? The fact that there may be another online subscription seems irrelevant here. The subscription has already been disconnected by this point by the dataplane status tracker.